lib/uicc_lwm2m/asn1_decode.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/util.h>
+
+#include "asn1_decode.h"
+
+bool asn1_dec_head(asn1_ctx_t *ctx, uint8_t *tag, size_t *len)
+{
+	uint32_t hlen = 2; /* Minimum two bytes for header */
+
+	if (ctx->error || ((ctx->offset + hlen) > ctx->length)) {
+		/* Error detected or out of data (happens at end of sequence) */
+		return false;
+	}
+
+	*tag = ctx->asnbuf[ctx->offset++];
+	*len = ctx->asnbuf[ctx->offset++];
+
+	if ((*tag & 0x1F) == 0x1F) {
+		/* Extended tag number is unsupported */
+		ctx->error = true;
+		return false;
+	}
+
+	if (*len & 0x80) {
+		int n = *len & 0x7F;
+
+		hlen += n;
+		if (n > 3 || (ctx->offset + hlen) > ctx->length) {
```

Comment:
so many magic numbers -> defines

Reply:
Added `ASN1_MAX_LENGTH_BYTES` instead of `3`.